### PR TITLE
fix(InputNumber): Fix incorrect disable state style for input box com…

### LINKED ISSUE #42878: InputNumber与Input disable的样式不统一
### Reproduction link

[https://codepen.io/sansheng741/pen/rNQNzdM?editors=0010](https://codepen.io/sansheng741/pen/rNQNzdM?editors=0010)

### Steps to reproduce

给InputNumber与Input添加前缀或者后缀和disabled=true，两个组件展现出来的样式不一致，InputNumber组件前后缀会高亮而Input组件不会。

### What is expected?

InputNumber的前后缀在添加disabled=true属性后，取消高亮

### What is actually happening?

InputNumber组件前后缀会高亮而Input组件不会

| Environment | Info |
| --- | --- |
| antd | 5.5.2 |
| React | 18 |
| System | window10 |
| Browser | Google Chrome 114.0.5735.90 |

### PR DESCRIPTION
**Pull request by biubiukam**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

1. fix #42878 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix incorrect disable state style for input box components with front and rear label configuration.      |
| 🇨🇳 Chinese |      修复具有前后标签配置的输入框组件的不正确禁用状态样式。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c2c57cc</samp>

This pull request enhances the `InputNumber` component by improving its TypeScript compatibility, disabled state handling, and addon style. It also fixes some bugs and inconsistencies with the `Form` component and undefined or null props. Due to the differences in DOM structure between the InputNumber component and Input, the class names for disabled states are independently set on the pre and post labels.

### <samp>🤖 Generated by Copilot at c2c57cc</samp>

* Initialize inputRef with null as HTMLInputElement to avoid TypeScript error ([link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL37-R37))
* Coerce controlsTemp to boolean value to avoid passing undefined or null to RcInputNumber ([link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL123-R123))
* Use mergedDisabled to respect disabled state from Form component ([link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL134-R134), [link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL175-R193), [link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857R132-R135))
* Add conditional className to addonBefore and addonAfter nodes based on mergedDisabled ([link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-6c85e13d1b03c66a69812b7717bd3695471bc2a1a381475e085c9dcf28662faaL175-R193))
* Update genInputNumberStyles to include style rule for disabled addon nodes ([link](https://github.com/ant-design/ant-design/pull/42944/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857R132-R135))

### CODE REVIEW

File: components/input-number/index.tsx [hunk 1]
```diff
@@ -172,10 +172,25 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
   if (hasAddon) {
     const wrapperClassName = `${prefixCls}-group`;
     const addonClassName = `${wrapperClassName}-addon`;
+    const addonDisabledClassName = `${addonClassName}-disabled`;
     const addonBeforeNode = addonBefore ? (
-      <div className={addonClassName}>{addonBefore}</div>
+      <div
+        className={classNames(addonClassName, {
+          [addonDisabledClassName]: mergedDisabled,
```
Comment (afc163) [inline components/input-number/index.tsx:179]: 外面已经有 ${prefixCls}-affix-wrapper-disabled 了，是否有必要再在这里加 disabled 的类名？

Comment (kampiu) [reply components/input-number/index.tsx:179]: 当存在前后置标签时，发现整个组件中不存在 ${prefixCls}-affix-wrapper-disabled，不过我对比了文档中另外一个disabled的场景后，我想我知道问题在哪里了，我重新调整下
E1EE3074-ABB5-4320-A52C-AA27A9709F57

Comment (kampiu) [reply components/input-number/index.tsx:179]: InputNumber的props中，`prefix`与`addonAfter`/`addonBefore`是分别用不同的className包裹，只有在`prefix`生效时才会有`${prefixCls}-affix-wrapper-disabled`。所以我打算将 `${prefixCls}-affix-wrapper-disabled`的类名往上提一层，放置在判断`prefix`与`addonAfter`/`addonBefore`之外。

**Comment by kampiu** [comment]
`prefix`与`addonAfter` / `addonBefore`的配置效果如下:
**before**:
image
**after**:
image

**Pull request closed without merging**